csrc/device_lower/validation.cpp

```diff
@@ -373,6 +373,16 @@ class VectorizeValidator : public OptInDispatch {
       if (r_id->isReduction() || r_id->isBroadcast()) {
         continue;
       }
+      if ((tv->getMemoryType() == MemoryType::Shared ||
+           tv->getMemoryType() == MemoryType::Local) &&
+          r_id->isBlockDim()) {
+        // Inner-most parallelized dimensions don't count in allocation of
+        // shared and local tensors.
+        continue;
+      }
```

Comment:
I am vectorizing the store to smem, but without this I hit 
> Vectorized dim for consumer has to be from a contiguous inner most position. tv: T13_s[ iblockIdx.x56{( ceilDiv(( (( (( getMetaData(T0) )).logical_size ))[0] ), 128) )}, iblockIdx.y58{( ceilDiv(( (( (( getMetaData(T1) )).logical_size ))[1] ), 128) )}, iblockIdx.z55{2}, ithreadIdx.z353{( ceilDiv(( ( ceilDiv(128, 4) ) * 4 ), 64) )}, ithreadIdx.y355{( ceilDiv(( ( ( ceilDiv(( ceilDiv(128, 8) ), 4) ) * 4 ) * 8 ), 64) )}, iS357{( ceilDiv(64, 16) )}, iS359{( ceilDiv(64, 8) )}, ithreadIdx.x370{( ( ( ceilDiv(( ceilDiv(16, 8) ), 2) ) * 8 ) * ( ceilDiv(8, 2) ) )}, iS366{( ceilDiv(8, 8) )}, iS364{2}, iV369{2} ] ca_pos( 2 ) produce_pos( 3 ), allocation domain: iS412{( (( (( getMetaData(T0) )).logical_size ))[0] )}, iS413{( (( (( getMetaData(T1) )).logical_size ))[1] )}, iblockIdx.z55{2}, vectorized id: iS413{( (( (( getMetaData(T1) )).logical_size ))[1] )}, innermost id: iblockIdx.z55{2}, contiguity: t

For a local or shared tensor, block-parallelized dimensions don't affect the allocation, so we should just skip them here I think.